### PR TITLE
gcp - add cloud dns initial setup

### PR DESCRIPTION
**Pull request by egor-miasnikov**
Added initial classes with `query` and `get` functionality for `Managed Zone` and `Policy` resources in Cloud DNS service

### CODE REVIEW

File: tools/c7n_gcp/c7n_gcp/resources/dns.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        scope = 'project'
+
+        @staticmethod
+        def get(client, resource_info):
+            return client.execute_query(
+                'get', {'project': resource_info['project_id'],
+                        'managedZone': resource_info['name']})
+
+
+@resources.register('policy')
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/dns.py]: change this to dns-policy

File: tools/c7n_gcp/c7n_gcp/resources/dns.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from c7n_gcp.provider import resources
+from c7n_gcp.query import QueryResourceManager, TypeInfo
+
+
+@resources.register('managed-zone')
```
Comment (bclements) [inline tools/c7n_gcp/c7n_gcp/resources/dns.py]: Change this to dns-managed-zone

File: tools/c7n_gcp/tests/test_dns.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        project_id = 'cloud-custodian'
+        session_factory = self.replay_flight_data(
+            'managed-zone-query', project_id=project_id)
+
+        policy = self.load_policy(
+            self.dryrun_policy,
+            session_factory=session_factory)
+
+        managed_zone_resources = policy.run()
+        self.assertEqual(len(managed_zone_resources), 1)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_dns.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

File: tools/c7n_gcp/tests/test_dns.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        project_id = 'cloud-custodian'
+        session_factory = self.replay_flight_data(
+            'policy-query', project_id=project_id)
+
+        policy = self.load_policy(
+            self.dryrun_policy,
+            session_factory=session_factory)
+
+        policy_resources = policy.run()
+        self.assertEqual(len(policy_resources), 1)
```
Comment (bclements) [inline tools/c7n_gcp/tests/test_dns.py]: We need to be more explicit about the results we expect to be returned. What I mean by this is if we expect 7 results we should assertEqual on the specific list of 7 results. 

So if you expect a policy run to return 7 results with the names "location1, location2, location3, location4, location5, location6, location7", then verify that is what you get back in your test.

**Comment by alexkarpitski** [comment]
@egor-miasnikov @bclements the suggested changes have been done.

**Pull request closed without merging**